internal/coreinternal/attraction/attraction.go

```diff
@@ -168,13 +170,21 @@ type attributeAction struct {
 // AttrProc is an attribute processor.
 type AttrProc struct {
 	actions []attributeAction
+	logger  *zap.Logger
 }
 
 // NewAttrProc validates that the input configuration has all of the required fields for the processor
 // and returns a AttrProc to be used to process attributes.
 // An error is returned if there are any invalid inputs.
-func NewAttrProc(settings *Settings) (*AttrProc, error) {
-	var attributeActions []attributeAction
+func NewAttrProc(settings *Settings, logger *zap.Logger) (*AttrProc, error) {
+	if logger == nil {
```

Comment:
I think now we might assume that logger is never nil?

Reply:
It's good to check for nil, but not sure about the outcome: you do want to have access to a logger, it's not an optional parameter, so, you should return an error if it's nil.

Reply:
I agree, this seems to be unrelated, consider to split a separate PR.